### ISSUE TITLE
Feature request: Improve sample stories UI

### ISSUE THREAD

**Issue by richardeschloss** [richardeschloss/nuxt-stories#30]
This may be a collection of sub-tasks for improving the UI that ships with the sample stories. I think I like some aspects of the storybook UI, but may come up other ideas/improvements that will be captured here. I can understand that people searching for "nuxt storybook" were hoping for storybook's UI, but I think I can create something similar and perhaps even better, without the bloat and complicated configuration steps. 

- Currently, the sample stories ship with bootstrap-vue. To keep things more lightweight, perhaps we may want to ship with no css framework (just use native css and achieve similar aesthetic).
- Sidebar nav could be a tree nav instead
- Stories could be edited directly on the UI instead of the IDE. I think this would improve the DX. 
- It may be desired to persist the stories in localStorage

**Comment by richardeschloss** [comment]
Things I **like** about the StorybookUI, that may get pulled into here:
* The tree nav for stories, and the fonts and glyphs used there.
* The sidebar resizable feature. 
* The sidebar search/filter feature
* The actions panel
* Keyboard shortcuts
* Controls for prev/next component and stories. 
* Control for copying the canvas link.

Things I **don't like**:
* The Storybook logo being visible. I would want to use that real estate for either more stories, or for the project name. I think logo could still be visible [somewhere] but in a better location. Like, maybe the sidebar footer would contain something like "Made with [heart] by Storybook". But, let the stories be king.
* The time it takes to build the storybook UI when I am using just 3 stories, using their tutorial. Usually things like this scare me off because I don't know how the slow the build time will get when I add, say, 100 stories. Build time needs to be faster.
* The "collapse all" feature isn't working on my end. Also, while there is a "collapse all", I don't see an option to perform the reverse "expand all".
* The topbar and it's controls. Full-window mode should make it mostly disappear, the way I do it. This will let users see what their site will look like in a more accurate form. While "open canvas in new tab" sorta does the job, it requires the user to jump back and forth between tabs, which I think can lead to poor UX. 
* The icon used in the top bar for "open canvas" *looks* like it would instead perform an action of "upload storybook" or "share storybook" which I think could be more useful feature.
* The zoom in / zoom out controls I think are overkill. Browser zoom seems to be fine, but I can understand browser zoom will zoom *everything*, not just the canvas. The canvas zoom would be useful if, after a user chooses a size preference, if that new size could easily be saved back in the stylesheets.

**Comment by richardeschloss** [comment]
Status Update:

* First, an apology for my delay on this. My efforts were primarily on [nuxt-socket-io](https://github.com/richardeschloss/nuxt-socket-io).
* I *am* actually playing around with some UI improvements in my own app! One thing I've discovered is that a lot of functionality *already exists* in Vue dev tools! Vue dev tools is technically the knobs, although, it could be easier to work with for some data types.
* I was also thinking recently that maybe the stories should auto-generate based on the Nuxt folder structure. This has pros and cons. Pro is: the nuxt folder structure is already known, just build out story skeletons for "components", "pages", "store" etc.  The con is: it may lock-in the UI into Nuxt. But it may be a loose lock-in and may not be so hard to specify a new stories structure. *maybe*..

**Comment by richardeschloss** [comment]
Just jotting some more notes here: I think updating the markdown component to accept a "src" property would make it even easier to work with. So, I was thinking it could be:

```
// For a whole directory:
<markdown src="/path/to/mdDir" />

// For a single file:
<markdown src="/path/to/mdFile" />
```
I've seen quite a bit of solutions, but I couldn't quite find a Vue component that works like this. If it exists, I'll be happy to use it. 

I was trying to get the markdown component to be a hybrid markdown/vue (which doesn't exactly work the way I intended)

**Comment by richardeschloss** [comment]
More note-jotting here: This may be a major change coming but I think it may be a good idea to use similar themes as the NuxtJS website? So, their website is completely based on tailwindcss, whereas the current nuxt-stories UI is based on bootstrap-vue (which can be buggy at times, I'm finding, even though bootstrap is what I know). I think to provide some sort of consistency with what the Nuxt community does, it may be a good idea to use their UI for the stories (and tweak it as needed). I like their navbar, their sidebar (but the sidebar looks like it could be limited to 2 levels? may or may not be a bad thing...)

**Comment by richardeschloss** [comment]
Also...if anyone is reading this, is their a preference: tailwind vs. bootstrap vs. native css? Bootstrap comes with utility classes that make it easy to style desktop vs. mobile layouts, but utility classes seems to be more of a priority in tailwind. I'm open to feedback.

**Comment by richardeschloss** [comment]
More notes: 
- It may be extremely useful to connect to [bit.dev](https://bit.dev/) OR...even better...

--> for VUE components, to pull from [awesome-vue](https://github.com/vuejs/awesome-vue#components--libraries)

--> for NUXT, pull from [awesome-nuxt](https://github.com/nuxt-community/awesome-nuxt)

It seems that [awesomejs.dev](https://awesomejs.dev) already pulls from those (see also: [their github](https://github.com/Akryum/awesomejs.dev))! That's really cool! So, it sees like we would just want to use their api. (Or at least refer people there)

**Comment by richardeschloss** [comment]
Also, consider evaluating:

* ChakraUI (vue/nuxt), new css framework  which has accessibility as a key focus. 
* H-Wiki, holochain powered Wiki
- https://blog.holochain.org/h-wiki-is-here-take-ownership-of-your-collective-knowledge/
- Twitter: https://twitter.com/holochain?lang=en
- Front-end: https://github.com/eyss/h-wiki-front (React-based... :/ )
- Back-end: https://github.com/eyss/h-wiki-back

**Comment by richardeschloss** [comment]
More thoughts: I've been playing with Nuxt content for docs in another project, and I think it's pretty polished. I think there can be a lot of benefit pulling in their styling and layouts into here. Then the nuxt stories will look-and-feel more like Nuxt.

**Comment by richardeschloss** [comment]
Status update: the UI is looking pretty awesome on my end. Now with more markdown support (and vue-in-markdown support) with live editing on the UI. Patience dear friends as I have to make sure things get tested and I accounted for most use cases.

**Comment by richardeschloss** [comment]
Status update: Some awesome cool things coming...works on my end, but it'll be MAJOR changes. Mostly likely a major version bump, but for what I believe to be the better. It'll take some time to test and explain things.

**Comment by richardeschloss** [comment]
Status update: I have a lot of cool stuff working, but it's one thing to have it work in my local setup, and it's another to ensure it works properly when it's deployed as a module. So I am still testing things.

**Comment by richardeschloss** [comment]
Status update: 
- Tested stories.modulue to 100% cov
- Tested markdown utils to 100% cov

Some other cool (really fast) things working too.I don't want to spoil the surprise just yet ;)

**Comment by richardeschloss** [comment]
Status Update!! Cool stuff pushed...tested... to be pulled [and released?] hopefully this week. Major changes. Serious overhaul, but it's really cool now.

**Comment by richardeschloss** [comment]
Merged #35
